storage/internal/benchmarks/client_pool.go

```diff
@@ -132,81 +136,82 @@ func initializeClientPools(ctx context.Context, opts *benchmarkOptions) func() {
 // Rotate through clients. This may mean certain clients get a larger workload
 // than others, if object sizes vary.
 func getClient(ctx context.Context, br benchmarkResult) *storage.Client {
-	if br.params.api == grpcAPI || br.params.api == directPath {
+	switch br.params.api {
+	case grpcAPI, directPath:
 		return gRPCClients.Get()
+	case jsonAPI:
+		return jsonClients.Get()
+	case xmlAPI:
+		return xmlClients.Get()
 	}
-	return httpClients.Get()
+	return nil
 }
 
 // mutex on starting a client so that we can set an env variable for GRPC clients
 var clientMu sync.Mutex
 
-type clientParams struct {
-	customClient                    bool // if false, all other params are ignored
-	writeBufferSize, readBufferSize int
-	connectionPoolSize              int
-}
+func initializeHTTPClient(ctx context.Context, writeBufferSize, readBufferSize int, json bool) (*storage.Client, error) {
+	opts := []option.ClientOption{}
 
-func initializeHTTPClient(ctx context.Context, p clientParams) (*storage.Client, error) {
-	if !p.customClient {
-		clientMu.Lock()
-		c, err := storage.NewClient(ctx)
-		clientMu.Unlock()
-		return c, err
-	}
+	if writeBufferSize != useDefault || readBufferSize != useDefault {
+		// We need to modify the underlying HTTP client
+		dialer := &net.Dialer{
+			Timeout:   30 * time.Second,
+			KeepAlive: 30 * time.Second,
+		}
 
-	dialer := &net.Dialer{
-		Timeout:   30 * time.Second,
-		KeepAlive: 30 * time.Second,
-	}
+		// These are the default parameters with write and read buffer sizes modified
+		base := &http.Transport{
```

Comment:
You can just grab https://pkg.go.dev/net/http#DefaultTransport, clone it, then make additional modifications as needed from the supplied opts